key.core/src/main/antlr4/KeYParser.g4

```diff
@@ -329,7 +329,8 @@ literals:
   | string_literal
 ;
 
-term: parallel_term;
```

Comment:
The entry point to the `term` was `parallel_term`, which is a sequence of terms seperated by `||`. 

This mean, that `a || b` would be a valid term, although the `||` is only allowed inside UPDATES. 

Here I changed it: Now the top-level term is an `equivalence_term` (`a <-> b <-> c`), and `update_term` points `parallel_term`. 

I think, this is also not quite right. The term `{ 2+2 } x` should be rejected by the grammar, but is currently allowed. The assignment in `elementary_update_term` should be mandatory, but I think it is required by matcher expression.

Reply:
It will (probably) be rejected later during AST resolution. If you can make the grammar clearer, sure!
Updates can be: Identifiers, assignments, parallel updates, empty. I believe, I also saw cases of strangely nested updates, that should be checked.

Reply:
Yeah, the `updateRules.key` is the problem. Here there are rules with problematic terms:

```
    applyOnParallel {
        \schemaVar \update u, u2, u3;

        \find({u}(u2 || u3))

        \replacewith({u}u2 || {u}u3)

        \heuristics(update_apply_on_update)
    };
```

Not sure how to solve this.

Reply:
I currently see no solution. We are back on the old track.